### ISSUE TITLE
Seeing websocket close connections

### ISSUE THREAD

**Issue by battlemidget** [juju/python-libjuju#120]
Cross referencing from my bug: https://github.com/conjure-up/conjure-up/issues/854

```
2017-04-29 14:10:20,815 [ERROR] conjure-up/openstack-novalxd - juju.model: model.py:641 - Error in watcher
Traceback (most recent call last):
  File "/snap/conjure-up/239/lib/python3.6/site-packages/juju/model.py", line 630, in _start_watch
    self.loop)
  File "/snap/conjure-up/239/lib/python3.6/site-packages/juju/utils.py", line 94, in run_with_interrupt
    result = [f.result() for f in done if f is not event_task]
  File "/snap/conjure-up/239/lib/python3.6/site-packages/juju/utils.py", line 94, in <listcomp>
    result = [f.result() for f in done if f is not event_task]
  File "/snap/conjure-up/239/lib/python3.6/site-packages/juju/client/facade.py", line 392, in wrapper
    reply = await f(*args, **kwargs)
  File "/snap/conjure-up/239/lib/python3.6/site-packages/juju/client/_client1.py", line 59, in Next
    reply = await self.rpc(msg)
  File "/snap/conjure-up/239/lib/python3.6/site-packages/juju/client/overrides.py", line 99, in rpc
    result =  await self.connection.rpc(msg, encoder=TypeEncoder)
  File "/snap/conjure-up/239/lib/python3.6/site-packages/juju/client/connection.py", line 237, in rpc
    result = await self.recv(msg['request-id'])
  File "/snap/conjure-up/239/lib/python3.6/site-packages/juju/client/connection.py", line 174, in recv
    return await self.messages.get(request_id)
  File "/snap/conjure-up/239/lib/python3.6/site-packages/juju/utils.py", line 63, in get
    raise value
  File "/snap/conjure-up/239/lib/python3.6/site-packages/juju/client/connection.py", line 182, in receiver
    loop=self.loop)
  File "/snap/conjure-up/239/lib/python3.6/site-packages/juju/utils.py", line 94, in run_with_interrupt
    result = [f.result() for f in done if f is not event_task]
  File "/snap/conjure-up/239/lib/python3.6/site-packages/juju/utils.py", line 94, in <listcomp>
    result = [f.result() for f in done if f is not event_task]
  File "/snap/conjure-up/239/lib/python3.6/site-packages/websockets/protocol.py", line 299, in recv
    raise ConnectionClosed(self.close_code, self.close_reason)
websockets.exceptions.ConnectionClosed: WebSocket connection is closed: code = 1006, no reason.
2017-04-29 14:10:20,820 [DEBUG] conjure-up/openstack-novalxd - events.py:18 - Setting Error
```

**Comment by petevg** [comment]
A few weeks ago, I was seeing disconnects due to timeouts, esp. against the JaaS controllers. That was fixed as of v 0.4.1.

We still may get disconnects just due to random network issues. I don't think that it's appropriate to try to reconnect from within python-libjuju, however. There isn't a way to pick up the websocket connection where you left off, and python-libjuju can't really tell which requests have been processed on the juju side, and which have been dropped.

@battlemidget I think that the best solution here is to catch the ConnectionClosed exception in conjure-up, either alerting the user that something went wrong, or trying to pick up on the deploy process where you think that you have lost the connection.

**Comment by battlemidget** [comment]
As far as I can tell this has been resolved with cory's latest code
